vectordb_orm/fields.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC
+from vectordb_orm.indexes import IndexBase
+from vectordb_orm.similarity import SimilarityMetric
+from typing import Any
+
+class BaseField(ABC):
+    """
+    BaseField is the superclass to all fields that require additional customization behavior. They
+    are specified as values for the typehints that otherwise populate the class, like:
+
+    ```
+    class MyModel:
+        embedding: np.array = EmbeddingField(dim=16)
+    ```
+
+    """
+    def __init__(self, default: Any):
+        self.default = default
+
+
+class PrimaryKeyField(BaseField):
+    def __init__(self, default: Any = None):
+        super().__init__(default=default)
```

Comment:
what happens when this is None?

Reply:
Technically nothing, because we will explicitly set the primary values either in-code or by relying on the backend to provide an auto-increment. But I agree this API is a bit confusing to end users since it advertises this `default` flag. Changing in the next PR to remove this keyword argument.

Reply:
This seems like a good use case for a [dataclass](https://docs.python.org/3/library/dataclasses.html) with the `field(default_factory=blah)` feature